### PR TITLE
update sqlitecpp to 3.2.0

### PR DESCRIPTION
**Pull request by paulocoutinhox**
Specify library name and version:  **sqlitecpp/3.2.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by uilianries** [comment]
@paulocoutinhox please, consider https://github.com/paulocoutinhox/conan-center-index/pull/8

**Comment by paulocoutinhox** [comment]
Done

**Review by uilianries — DISMISSED** [review]
LGTM

**Comment by paulocoutinhox** [comment]
@uilianries is has an error:

```
Error: recipes/sqlitecpp/all/conanfile.py:71: [E1101(no-member), SQLiteCppConan._patch_sources] Instance of 'SQLiteCppConan' has no '_source_subfolder' member. Please, check https://github.com/conan-io/conan-center-index/blob/master/docs/v2_linter.md
```

Because of:

```
replace_in_file(self,
                os.path.join(self._source_subfolder, "include", "SQLiteCpp", "Utils.h"),
                "const nullptr_t nullptr = {};",
                "")
```

These lines was removed:

```
@property
def _source_subfolder(self):
    return "source_subfolder"
```

It is correct?

**Comment by paulocoutinhox** [comment]
I fixed it @uilianries ...can you check?

**Review by uilianries — APPROVED** [review]
Agora tá certo! LGTM

**Comment by paulocoutinhox** [comment]
@SpaceIm and @czoido can you approve here?

**Review by czoido — APPROVED** [review]

File: recipes/sqlitecpp/all/conanfile.py [hunk 1]
```diff
@@ -26,88 +30,84 @@ class SQLiteCppConan(ConanFile):
...
         self._patch_sources()
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        self.copy(pattern="LICENSE.txt", dst="licenses", src=self.source_folder)
```
Comment (SpaceIm) [inline recipes/sqlitecpp/all/conanfile.py:96]:
```suggestion
        copy(self, "LICENSE.txt", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
```

File: recipes/sqlitecpp/all/conanfile.py [hunk 2]
```diff
@@ -26,88 +30,84 @@ class SQLiteCppConan(ConanFile):
...
+                pass
 
     def requirements(self):
-        self.requires("sqlite3/3.38.5")
+        self.requires("sqlite3/3.39.3")
 
     def validate(self):
-        if tools.Version(self.version) >= "3.0.0" and self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, 11)
+        if Version(self.version) >= "3.0.0" and self.settings.compiler.get_safe("cppstd"):
```
Comment (SpaceIm) [inline recipes/sqlitecpp/all/conanfile.py:52]:
```suggestion
        if Version(self.version) >= "3.0.0" and self.info.settings.compiler.get_safe("cppstd"):
```

File: recipes/sqlitecpp/all/conanfile.py [hunk 3]
```diff
@@ -26,88 +30,84 @@ class SQLiteCppConan(ConanFile):
...
+            check_min_cppstd(self, 11)
         if self.settings.os == "Windows" and self.options.shared:
```
Comment (SpaceIm) [inline recipes/sqlitecpp/all/conanfile.py:54]:
```suggestion
        if self.info.settings.os == "Windows" and self.info.options.shared:
```

**Comment by SpaceIm** [comment]
Ok too late ^^addressed in https://github.com/conan-io/conan-center-index/pull/13044

**Pull request merged**